### DISCUSSION TITLE
Combining `sse` and `oob`

### DISCUSSION THREAD

**Discussion by fliepeltje** [bigskysoftware/htmx#2104] [Q&A]
I'd like to keep the element replacement logic on the server side, so ideally I want to send `oob` swaps from the server and on certain events. Is there a way of achieving this? I understand it is possible to use `hx-trigger` to manipulate content in this style, however I prefer using the oob swaps in most cases and that gets me into messy code when I have `innerHTML` and `outerHTML` versions of the same snippet. 

Alternatively, is there a way of registering new elements to listen to events when you stream them through sse? For instance, when I replace an element with a `hx-trigger` with a new sse event, it will nto listen for it (which is the main reason I end up having to do acrobatics in the first place :sweat_smile: )

**Comment by shepmaster** [comment]
Having SSE support out-of-band swaps would also make the SSE extension more in line with [the WebSocket extension](https://htmx.org/extensions/web-sockets/):

> Content that is sent down from the websocket will be parsed as HTML and swapped in by the id property, using the same logic as [Out of Band Swaps](https://htmx.org/attributes/hx-swap-oob/).

**Comment by 7flash** [comment]
I need this also, in my case i have element listening to its status with sse and once its completed should trigger a notification

**Comment by jvcop** [comment]
I second this, support for out-of-bands swaps would be great!

**Comment by jackielii** [comment]
Is this implemented? I always thought this is possible, turns out not to be.

**Comment by sopa0** [comment]
hi guys
created a pull request that does just that, just stole the code from the ws extension, feel free to give it a try

https://github.com/bigskysoftware/htmx-extensions/pull/94
